test/unit/components/navpatientheader/index.test.js

```diff
@@ -0,0 +1,298 @@
+        expect(wrapper.find('button#navPatientHeader_viewDataButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_profileButton').exists()).to.be.true;
+
+        expect(wrapper.find('button#navPatientHeader_shareButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_uploadButton').exists()).to.be.false;
+      })
+    });
+
+    context('personal user with root permissions', () => {
+      const props = { // personal user with root permissions;
+        user: { roles: [] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: true } },
+      };
+
+      const wrapper = mount(
+        <BrowserRouter>
+          <NavPatientHeader {...props} />
+        </BrowserRouter>
+      );
+
+      it('shows the correct info and actions', () => {
+        expect(wrapper.text()).to.include('Vasyl Lomachenko');
+        expect(wrapper.text()).not.to.include('567890');
+        expect(wrapper.text()).not.to.include('2010-10-20');
+
+        expect(wrapper.find('button#navPatientHeader_backButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_viewDataButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_profileButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_shareButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_uploadButton').exists()).to.be.true;
+      })
+    });
+
+    context('clinician user with non-root permissions', () => {
+      const props = { // personal user with root permissions;
+        user: { roles: ['clinician'] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: false } },
+      };
+
+      const wrapper = mount(
+        <BrowserRouter>
+          <NavPatientHeader {...props} />
+        </BrowserRouter>
+      );
+
+      it('shows the correct info and actions', () => {
+        expect(wrapper.text()).to.include('Vasyl Lomachenko');
+        expect(wrapper.text()).to.include('567890');
+        expect(wrapper.text()).to.include('2010-10-20');
+
+        expect(wrapper.find('button#navPatientHeader_backButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_viewDataButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_profileButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_shareButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_uploadButton').exists()).to.be.false;
+      })
+    });
+
+    context('clinician user with root permissions', () => {
+      const props = { // personal user with root permissions;
+        user: { roles: ['clinician'] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: true } },
```

Comment:
The much more likely scenario is that there's a clinician user with `upload` permissions. - `permissions: { upload: {} }`

Reply:
Gotcha, I see it now in the props edba060d63fc21252cb3ffd1f7f2ad3debe58476